Add a property to control scheduler timeout

Mr-Pine opened the pull request:
This defines a `timeout_ms` property that controls the `.timeout_ms` field of the scheduler `SCX_OPS`.
To support this, the property collection code in the compiler plugin has been fixed to always fallback to the default value. Previously this was only done when *none* of the properties were specified.

parttimenerd commented:
Please set the timeout in one of the scheduler samples. Every new feature should ideally be used somewhere.

Testing would also be nice, but this is probably more effort.

Mr-Pine commented:
I added the property to the minimal scheduler sample and added a test that checks for a timeout of 3 seconds, that the scheduler is unloaded between 2 and 4 seconds
